mmv1/third_party/terraform/tpgresource/utils.go

```diff
@@ -238,6 +238,39 @@ func ExpandStringMap(d TerraformResourceData, key string) map[string]string {
 	return ConvertStringMap(v.(map[string]interface{}))
 }
 
+// ExpandStoragePoolUrl
+func ExpandStoragePoolUrl(v interface{}, d TerraformResourceData, config *transport_tpg.Config) string {
```

Comment:
Since this is specific to compute, it should live in `services/compute` instead of in the top-level utils.

Please also add some [unit tests](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-unit-tests) for this function. (That might be easier to do if you extract the core logic of constructing the URL and building the string with templating, leaving the ReplaceVars call in the actual expander(s).